src/WebJobs.Script/Workers/Rpc/Configuration/RpcWorkerConfigFactory.cs

```diff
@@ -116,6 +120,17 @@ internal void AddProvider(string workerDir)
             {
                 try
                 {
+                    // After specialization, load worker config only for the specified runtime unless it's a multi-language app.
+                    if (!string.IsNullOrWhiteSpace(_workerRuntime) && !_environment.IsPlaceholderModeEnabled() && !_environment.IsMultiLanguageRuntimeEnvironment())
+                    {
+                        string workerRuntime = Path.GetFileName(workerDir);
+                        // We do not want to skip non-worker directories like function app payload directory.
```

Comment:
Message is a little confusing to me, I think this makes a little more sense - thoughts?

`Skip unrelated worker directories, but don’t skip things outside of worker directories (e.g., function app payload dirs), even if the runtime doesn't match.`

or 

// Only skip worker directories that don't match the current runtime.
// Do not skip non-worker directories like the function app payload directory.

Reply:
Switched to the second version.